sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/internal/state/MeterSharedState.java

```diff
@@ -104,20 +103,20 @@ public final void registerLongAsynchronousInstrument(
             .getViewRegistry()
             .findViews(instrument, getInstrumentationLibraryInfo());
     for (View view : views) {
+      AsynchronousMetricStorage currentStorage =
+          AsynchronousMetricStorage.longAsynchronousAccumulator(
+              view,
+              instrument,
+              meterProviderSharedState.getResource(),
+              getInstrumentationLibraryInfo(),
+              meterProviderSharedState.getStartEpochNanos(),
+              metricUpdater);
       // TODO - move this in a better location.
-      if (view.getAggregation().config(instrument) == null) {
+      if (currentStorage == null) {
```

Comment:
any thoughts on using `Optional` for this return type, rather than using `null` as the signal for no metrics? Or a null-object pattern for the storage?

Reply:
It's a great suggestion.  I had the same thought while writing this code.  I'll take a crack at it.

Right now the "null-object" I'm using is a aggregate storage with an empty list of backing aggregations/metrics.   We can do better.

Reply:
Attempted to use the null-object pattern (in all locations I could).  I rushed it a bit, so I sent it up for you to look at and I'm going to overnight it myself and look at it fresh in the morning to understand what I did that I'm not happy with (something is off, but it's no longer using `null`).

Reply:
Hmm, still don't really like how this looks, but I think it's good enough (and important aspects are behind `internal`)